### ISSUE TITLE
Attempt to upgrade to >=4.0.6 from 4.0.5: Curious ReferenceErrors regarding global variables

### ISSUE THREAD

**Issue by DorianGrey** [frankwallis/plugin-typescript#120]
I've tried to update this plugin from 4.0.5 to 4.0.7 yesterday and faced an extremely curious issue regarding type definition files which no longer seem to be properly included.

The project contains a `typings.json` with the following references:
```
{
  "version": false,
  "dependencies": {
    "lodash": "registry:npm/lodash#4.0.0+20160305082308",
    "moment": "registry:npm/moment#2.10.5+20160211003958"
  },
  "ambientDependencies": {
    "angular": "registry:dt/angular#1.5.0+20160320081937",
    "angular-ui-router": "registry:dt/angular-ui-router#1.1.5+20160317120654",
    "highcharts": "registry:dt/highcharts#4.1.9+20160316155526",
    "jquery": "registry:dt/jquery#1.10.0+20160316155526",
    "rangy": "registry:dt/rangy#0.0.0+20160316155526"
  }
}
```
The generated `browser.d.ts` gets included in a file called `shim.ts`, which also references a project-local type definition `jsRoutes.d.ts`:
```
///<reference path="../../../../typings/browser.d.ts"/>
///<reference path="./../../definitions/jsRoutes.d.ts"/>
```
This `shim.ts` is referenced by all relevant .ts files in the project.
Using a `tsconfig.json` like
```
{
  "files": [
    "typings/browser.d.ts"
  ],
  "compilerOptions": {
    "target": "es5"
  }
}
```
with plugin-typescript configured like the examples illustrate, this setup should provide proper type checking when bundling with `jspm.Builder#bundle`. And, furthermore, it should ignore non-Typescript-files. Unfortunately, it doesn't when using version 4.0.6 or 4.0.7, instead it causes reference errors like:
```
TypeScript { [Error: ReferenceError: angular is not defined
	  at execute (evalmachine.<anonymous>:20:13)
	  at run ([...]/node_modules/karma/node_modules/core-js/modules/es6.promise.js:89:22)
	  at [...]/node_modules/karma/node_modules/core-js/modules/es6.promise.js:102:28
	  at flush ([...]/node_modules/karma/node_modules/core-js/modules/_microtask.js:14:5)
	  at nextTickCallbackWith0Args (node.js:420:9)
	  at process._tickCallback (node.js:349:13)
	
	Error loading [...]/assets/javascripts/modules/common/components.module.ts] originalErr: [ReferenceError: angular is not defined] }
```
(where components.module.ts references the shim.ts mentioned above)
```
TypeScript { [Error: ReferenceError: window is not defined
	  at evalmachine.<anonymous>:30866:4
	  at Object.exports.runInThisContext (vm.js:54:17)
	  at run ([...]/node_modules/karma/node_modules/core-js/modules/es6.promise.js:89:22)
	  at [...]/node_modules/karma/node_modules/core-js/modules/es6.promise.js:102:28
	  at flush ([...]/node_modules/karma/node_modules/core-js/modules/_microtask.js:14:5)
	  at nextTickCallbackWith0Args (node.js:420:9)
	  at process._tickCallback (node.js:349:13)
	
	Evaluating [...]assets/lib/angular/angular.js
	Error loading [...]assets/javascripts/modules/common/providers.module.ts] originalErr: [ReferenceError: window is not defined] }
```
(same referencing behavior here).

I'm not sure what may cause these errors (especially the latter one should NEVER happen - angular.js is not a typescript file!), since everything workes fine with version 4.0.5. 
I'd guess it has something to do with [this](https://github.com/frankwallis/plugin-typescript/commit/58cc11c5d0cebf91ba8d00e881fbde1e1b4c948a) commit, but I'm not sure why.

jspm version: `0.17.0-beta.13` (did not work with `0.17.0-beta.12` either)

Any clues what might be the cause of this curious behavior?

**Comment by frankwallis** [comment]
It does sound rather strange, thanks for the detailed report, and I'm sorry I broke your build :(

I haven't been able to recreate it, but I have a couple of questions:

- You say that ```shim.ts``` is referenced by all relevant .ts files, is this with a reference tag or an import?
- If you go into ```jspm_packages/github/frankwallis/plugin-typescript@4.0.7/plugin.js``` and change line 44 to be ```System.load``` instead of ```System.import``` does that change anything? I am hoping that this will stop SystemJS from trying to evaluate the module which I think is what is causing the error (it is a runtime error by the look of it).

**Comment by DorianGrey** [comment]
> and I'm sorry I broke your build
It's better to break a dev build than a production one ;)

`shim.ts` essentially contains something like:
```
///<reference path="../../../../typings/browser.d.ts"/>
///<reference path="./../../definitions/jsRoutes.d.ts"/>

import "./utils/consolePolyfill";
import "es5-shim";
import "jquery";
import "angular";
import "moment";
import "moment-timezone";
[...]
```
i.e. it picks up the references and initalizes the import of some libs affecting the global scope (most of them are still managed by bower, didn't have the time yet to move this handling to JSPM) to ensure they get loaded in the correct order.
From a technical point of view, our application has several entry points, but all of them share common code, introduced by a `common.module.ts`, which imports `shim.ts` (using the `import` statement) and itself gets imported by all entry point definitions. I.e. the references to `shim.ts` are - in most cases - transitive. That did work fine up to including v4.0.5 of your plugin, respectively [this](https://github.com/frankwallis/plugin-typescript/commit/58cc11c5d0cebf91ba8d00e881fbde1e1b4c948a) commit.
I've already tried to put the reference paths in the other files manually, but that did not change anything regardings the errors. 

I'm afraid that the same goes to changing `System.import` to `System.load` within `plugin.js` :(

**Comment by frankwallis** [comment]
FYI - I can now recreate this so now I just need to work out how to fix it... I'm going to move this over to https://github.com/frankwallis/plugin-typescript/issues/97 as it is a continuation of that thread.

**Comment by frankwallis** [comment]
This should now be fixed in 4.0.8

**Comment by DorianGrey** [comment]
Works fine now, thank you!
